src/main/java/com/google/alpollo/AnalysisHelper.java

```diff
@@ -128,4 +137,24 @@ public static String getInterpretation(double score, double magnitude) {
 
     return MIXED;
   }
+
+  public static List<SongEntity> filterDuplicates(List<SongEntity> list) {
+    HashMap<String, SongEntity> map = new HashMap<>(); 
+
+    for (SongEntity entity : list) {
+      if (map.containsKey(entity.getName())) {
+        SongEntity existingEntity = map.get(entity.getName());
+        Set<String> newTypes = entity.getType();
+        Set<String> existingTypes = existingEntity.getType();
+        double newSalience = entity.getSalience() + existingEntity.getSalience();
+        
+        existingTypes.addAll(newTypes);
+        existingEntity.setSalience(newSalience);
+      } else {
+        map.put(entity.getName(), entity);
+      }
+    }
+
+    return new ArrayList<SongEntity>(map.values());
```

Comment:
I think there are too many list creations in this whole story, which is not great for performance. Here you create a list from the map.values() collection. But as far as I can tell, this list is only used as an input to getTopSalientEntities, which makes a copy of its input list as the first thing it does.
At least one of those copies seems unnecessary

Reply:
Yep you're right. Sincerely, I have no idea why I'm making a copy in getTopSalientEntities().

Reply:
I think you were copying it to avoid modifying the list passed as a parameter. I personally wouldn't expect a function called `getSomethingBlaBla(myList)` to modify `myList`.

If we're only using these two functions together, the best thing to do is probably merge them (remember to rename the function accordingly). So instead of returning `new ArrayList<SongEntity>(map.values())` here you'd already sort and return the top entities. Something like the code below:

``` java
return map.values().stream()
    .sorted(SongEntity.ORDER_BY_SALIENCE_DESCENDING)
    .limit(MAX_ENTITIES)
    .collect(Collectors.toList());
```